src/__tests__/error-reporting.test.ts

```diff
@@ -27,14 +27,28 @@ describe("tests with definitions containing errors to see what's reported", () =
             expected_errors: [
                 {
                     "Error code": StateMachineErrorCode.SchemaValidationFailed,
-                    Message: "/States/PassState is invalid.",
+                    Message: "/States/PassState is invalid. must match exactly one schema in oneOf",
                     schemaError: {
                         instancePath: "/States/PassState",
                         schemaPath: "#/oneOf"
                     }
                 }
             ]
+        },
+        {
+            file: "invalid-payload-template.asl.json",
+            expected_errors: [
+                {
+                    "Error code": StateMachineErrorCode.SchemaValidationFailed,
+                    Message: "/States/Hello, World/Parameters/lorem.$ is invalid. must match format \"asl_payload_template\"",
+                    schemaError: {
+                        instancePath: "/States/Hello, World/Parameters/lorem.$",
+                        schemaPath: "#/oneOf/0/patternProperties/%5E.%2B%5C.%5C%24%24/format"
```

Comment:
Could you add a comment to explain what’s happening here with %5E.%2B%5C.%5C%24%24 ?

Reply:
the schema in this case is defined as a patternProperties and the key is a regex that needs escaping as a path.

I left a comment in the src about it. Not sure we need to decode this for display but we could with a url decoder.

Reply:
Thanks